core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -265,6 +266,24 @@ object DataMirror {
       * @param target BaseModule to get IOs from
       */
     def currentModulePorts(target: BaseModule): Seq[Data] = target.getIOs
+
+    /** Returns the current ports of an Instance.
+      *
+      * This method does not necessarily return the final ports of the target Instance. It consults Chisel's internal
+      * data structures to extract the module's IOs. For this reason, it is generally not safe, and users should prefer
+      * `DataMirror.modulePorts`, but this method may be used for certain use cases that want the current list of
+      * ports before the module is closed.
+      *
+      * @param target Instance to get IOs from
+      */
+    def currentInstancePorts(target: Instance[BaseModule]): Seq[Data] = {
+      target.underlying match {
+        case Clone(clone: ModuleClone[_]) =>
+          clone.getPorts.elements.map(_._2).toSeq
```

Comment:
```suggestion
          clone.getPorts.elementsIterator.toSeq
```
This saves an iteration and is inline with my medium-term goal of changing `elements` to be a `def` which will make calling it expensive.

Reply:
Got it, will do. I always forget I can use these APIs from within Chisel.